### PR TITLE
[12.0][ADD]product_creation_prevention

### PR DESCRIPTION
**Pull request by dsolanki-initos**
FYI @jans23

### CODE REVIEW

File: product_creation_prevention/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (jans23) [inline product_creation_prevention/__manifest__.py]: remove it.

**Comment by jans23** [comment]
Please rebase to the latest 12.0 because the older 12.0 had an issue which prevents me from testing.

File: product_creation_prevention/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from . import models
```
Comment (manengel) [inline product_creation_prevention/__init__.py]: Please add a newline here. CI will otherwise complain about it at some point.

**Comment by manengel** [comment]
@jans23 Did you by any chance mean this error: `
Odoo Server Error
Traceback (most recent call last):
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 656, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 314, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/srv/odoo/odoo/parts/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 698, in dispatch
    result = self._call_function(**self.params)
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 346, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/srv/odoo/odoo/parts/odoo/odoo/service/model.py", line 98, in wrapper
    return f(dbname, *args, **kwargs)
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "/srv/odoo/odoo/parts/odoo/addons/web/controllers/main.py", line 967, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/srv/odoo/odoo/parts/odoo/addons/web/controllers/main.py", line 959, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/srv/odoo/odoo/parts/odoo/odoo/api.py", line 755, in call_kw
    return _call_kw_model(method, model, args, kwargs)
  File "/srv/odoo/odoo/parts/odoo/odoo/api.py", line 728, in _call_kw_model
    result = method(recs, *args, **kwargs)
  File "/srv/odoo/odoo/parts/odoo/odoo/models.py", line 1396, in load_views
    for [v_id, v_type] in views
  File "/srv/odoo/odoo/parts/odoo/odoo/models.py", line 1396, in <dictcomp>
    for [v_id, v_type] in views
  File "/srv/odoo/odoo/parts/OCA/social/base_search_mail_content/models/mail_thread.py", line 47, in fields_view_get
    submenu=submenu)
  File "/srv/odoo/odoo/parts/odoo/addons/mail/models/mail_thread.py", line 409, in fields_view_get
    res = super(MailThread, self).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
  File "/srv/odoo/odoo/parts/odoo/odoo/models.py", line 1482, in fields_view_get
    xarch, xfields = View.postprocess_and_fields(self._name, etree.fromstring(result['arch']), view_id)
  File "/srv/odoo/odoo/parts/odoo/odoo/addons/base/models/ir_ui_view.py", line 1138, in postprocess_and_fields
    self.raise_view_error(message, view_id)
  File "/srv/odoo/odoo/parts/odoo/odoo/addons/base/models/ir_ui_view.py", line 568, in raise_view_error
    raise ValueError(message)
ValueError: Field `freetext` does not exist

Error context:
View `n/a`
[view_id: n/a, xml_id: n/a, model: n/a, parent_id: n/a]
`

**Review by manengel — CHANGES_REQUESTED** [review]
@dsolanki-initos Please apply a code formatter on this. @jans23 In general, I am rather critical about this module. Like, I get you want to prevent product creation. But I am not sure if this is what you really want: 
* It would prevent ANY product creation, for anybody as I get it. Does not matter for whom
* Odoo will prevent it in a graphical way, but will NOT enforce it on a server-side level, because permission-wise, whoever finds a way around that will still be theoretically allowed to create a product. 
* In order to make Odoo enforce it security-wise, I would definitely recommend the creation of a ir.model.access.csv file, so that if anybody ever attempts to trick Odoo in some way will eventually not succeed most likely.

**Comment by jans23** [comment]
> @jans23 Did you by any chance mean this error: `Odoo Server Error Traceback (most recent call last): File "/srv/odoo/ir_ui_view.py", line 568, in raise_view_error raise ValueError(message) ValueError: Field`freetext` does not exist
> 
> Error context: View `n/a` [view_id: n/a, xml_id: n/a, model: n/a, parent_id: n/a] `

This field "freetext" is defined in nitrokey_customization module. I don't see a reason why this field relates to this module but this error seem to be proof a dependency. @dsolanki-initos Can this technical dependency be removed?

**Comment by jans23** [comment]
> * In order to make Odoo enforce it security-wise, I would definitely recommend the creation of a ir.model.access.csv file, so that if anybody ever attempts to trick Odoo in some way will eventually not succeed most likely.

Yes, this sounds like a better approach. @dsolanki-initos Could you make this configurable (in user settings)?

**Comment by manengel** [comment]
> This field "freetext" is defined in nitrokey_customization module. I don't see a reason why this field relates to this module but this error seem to be proof a dependency. @dsolanki-initos Can this technical dependency be removed?

Uhm, just to clarify. I ran into this error when I wanted to prepare my testing environment with a fresh Odoo 12 copy having the modules installed there for our project. This module in particular had no involvement in that.

**Comment by jans23** [comment]
> Uhm, just to clarify. I ran into this error when I wanted to prepare my testing environment with a fresh Odoo 12 copy having the modules installed there for our project. This module in particular had no involvement in that.

Please report it at Gitlab.

**Comment by jans23** [comment]
@manengel Please have another look.

**Review by manengel — APPROVED** [review]

**Pull request merged**